backend/src/test/java/de/neuefische/capstone/backend/ProjectServiceTest.java

```diff
@@ -105,4 +103,62 @@ void returnListOfProjects(){
         assertEquals(expectedProjectList, actualProjectList);
     }
 
+    @Test
+    void whenProjectUpdated_thenReturnUpdatedProject() {
+        //Given
+        String id = "01A";
+        ProjectWithoutId projectWithoutId = new ProjectWithoutId(
+                "Earthquake Turkey",
+                "Help for the people in Turkey",
+                Category.PARTICIPATION,
+                List.of(Demand.DONATIONINKIND, Demand.MONEYDONATION),
+                50,
+                "Turkey");
+
+        Project expectedProject = new Project(
+                "01A",
+                projectWithoutId.name(),
+                projectWithoutId.description(),
+                projectWithoutId.category(),
+                projectWithoutId.demands(),
+                projectWithoutId.progress(),
+                projectWithoutId.location());
+
+        //When
+        when(projectRepo.save(expectedProject))
+                .thenReturn(expectedProject);
+        when(projectRepo.existsById(id))
+                .thenReturn(true);
+
+        Project actualProject = projectService.updateProject("01A", projectWithoutId);
+
+
+        //Then
+        verify(projectRepo).save(expectedProject);
+        assertEquals(expectedProject, actualProject);
+    }
+
+    @Test
+    void whenNoId_thenThrowException() {
+        //Given
+        String id = "01A";
+        ProjectWithoutId projectWithoutId = new ProjectWithoutId(
+                "Earthquake Turkey",
+                "Help for the people in Turkey",
+                Category.PARTICIPATION,
+                List.of(Demand.DONATIONINKIND, Demand.MONEYDONATION),
+                50,
+                "Turkey");
+
+
+        //When
+        when(projectRepo.existsById("01A"))
+                .thenReturn(false);
+
+
+        //Then
+        assertThrows(NoSuchElementException.class, () -> projectService.updateProject(id, projectWithoutId));
+        verify(projectRepo).existsById(id);
```

Comment:
Just a small suggestion: 
According to coaches the verify should be before the assertions because if the assertation passes then the verify won't be checked. So we should first place the verify and after that the assertations.

Reply:
Your absolutely right! 
But in this case, the repo call first happens inside the assertThrows statement. When putting the verify before assertThrow the test fails and stops at the verify statement :)

Reply:
Ohh I see, interesting! Learned something new 😎
Keep up the great work!